### PR TITLE
Components and Props Translated

### PR DESCRIPTION
**Pull request by zrakhimov**
Hello 👋

Translated "Components and Props" section to Uzbek. Please review and let me know if any suggestions or questions.

Thanks!

image

### CODE REVIEW

**Comment by zrakhimov** [comment]
@doniyor2109 @umidbekk @turdiyev

File: content/docs/components-and-props.md [hunk 1]
```diff
@@ -16,57 +16,57 @@ prev: rendering-elements.html
...
+Yuqoridagi ikki komponentlar React nuqtai nazaridan bir biriga teng.
 
-Function and Class components both have some additional features that we will discuss in the [next sections](/docs/state-and-lifecycle.html).
+[Keyingi bo'limlarda](/docs/state-and-lifecycle.html) funksiya va klass komponentlarini o'ziga hos qo'shimcha xususiyatlari haqida muhokama qilamiz.
 
-## Rendering a Component {#rendering-a-component}
+## Komponentlarni Chizish {#rendering-a-component}
 
-Previously, we only encountered React elements that represent DOM tags:
+Bundan oldin biz faqat HTML teglariga ega bo'lgan React elementlarini ko'rib chiqgan edik:
```
Comment (doniyor2109) [inline content/docs/components-and-props.md]:
```suggestion
Bundan oldin biz faqat DOM teglariga ega bo'lgan React elementlarini ko'rib chiqgan edik:
```

File: content/docs/components-and-props.md [hunk 2]
```diff
@@ -16,57 +16,57 @@ prev: rendering-elements.html
...
 
-However, elements can also represent user-defined components:
+Ammo, React elementlar dasturchi tomonidan yaratilgan komponentlarni ham o'z ichiga olishi mumkin
 
 ```js
 const element = <Welcome name="Sara" />;
 ```
 
-When React sees an element representing a user-defined component, it passes JSX attributes and children to this component as a single object. We call this object "props".
+Agarda React dasturchi yaratgan komponentni element orqali uchratsa, JSX attributlari va bolalarini shu komponentga bir dona obyekt sifatida uzatadi. Biz bu obyektni "props" deb ataymiz.
```
Comment (doniyor2109) [inline content/docs/components-and-props.md]:
```suggestion
Qachonki React dasturchi yaratgan komponentni element ko'rinishida uchratsa, JSX attributlari va bolalarini shu komponentga bir dona obyekt sifatida uzatadi. Biz bu obyektni "props" deb ataymiz.
```

File: content/docs/components-and-props.md [hunk 3]
```diff
@@ -78,28 +78,28 @@ ReactDOM.render(
...
-2. React calls the `Welcome` component with `{name: 'Sara'}` as the props.
-3. Our `Welcome` component returns a `<h1>Hello, Sara</h1>` element as the result.
-4. React DOM efficiently updates the DOM to match `<h1>Hello, Sara</h1>`.
-
->**Note:** Always start component names with a capital letter.
+>**Eslatma:** Komponent nomini doim katta harfdan boshlang.
 >
->React treats components starting with lowercase letters as DOM tags. For example, `<div />` represents an HTML div tag, but `<Welcome />` represents a component and requires `Welcome` to be in scope.
+>React kichik harf bilan boshlangan komponentlarni DOM teglari sifatida ko'radi.   
+Masalan,`<div />` HTML div teg ni belgilaydi, lekin `<Welcome />` esa komponentni belgilaydi va kod qamrovida bo'lishini talab qiladi.
```
Comment (doniyor2109) [inline content/docs/components-and-props.md]:
```suggestion
Masalan,`<div />` HTML div teg ni ifodalaydi, lekin `<Welcome />` esa komponentni ifodalaydi va `Welcome` ni shu kodning qamrovida bo'lishini talab qiladi.
```

File: content/docs/components-and-props.md [hunk 4]
```diff
@@ -213,7 +213,7 @@ function UserInfo(props) {
 }
 ```
 
-This lets us simplify `Comment` even further:
+Bu esa `Comment` qismini bundanda soddalashtirishga imkon beradi:
```
Comment (doniyor2109) [inline content/docs/components-and-props.md]:
```suggestion
Bu esa `Comment` ni bundanda soddalashtirishga imkon beradi:
```

File: content/docs/components-and-props.md [hunk 5]
```diff
@@ -233,30 +233,30 @@ function Comment(props) {
 
 [](codepen://components-and-props/extracting-components-continued)
 
-Extracting components might seem like grunt work at first, but having a palette of reusable components pays off in larger apps. A good rule of thumb is that if a part of your UI is used several times (`Button`, `Panel`, `Avatar`), or is complex enough on its own (`App`, `FeedStory`, `Comment`), it is a good candidate to be extracted to a separate component.
+Komponentlarni ajratish boshida mayda ishga o'xshab tuyilishi mumkin, lekin katta ilovalarda qayta ishlatiladigan komponentlar to'plamiga ega bo'lish ko'p yengilliklar tug'diradi. Agarda sizning UI'ingizdagi qism bir necha marta ishlatiladigan bo'lsa (`Button`, `Panel`, `Avatar`) yoki o'z o'zidan murakkab bo'lsa (`App`, `FeedStory`, `Comment`), komponentga ajratish uchun yaxshi nomzod bo'ladi.
```
Comment (doniyor2109) [inline content/docs/components-and-props.md]:
```suggestion
Komponentlarni ajratish boshida mayda ishga o'xshab tuyilishi mumkin, lekin katta ilovalarda qayta ishlatiladigan komponentlar to'plamiga ega bo'lish ko'p yengilliklarni yaratadi. Agarda sizning UI'ingizdagi qaysidir bo'lak bir necha marta ishlatiladigan bo'lsa (`Button`, `Panel`, `Avatar`) yoki o'z o'zidan murakkab bo'lsa (`App`, `FeedStory`, `Comment`), shu bo'lakni komponentga ajratish yaxshi nomzod bo'ladi. 
```

File: content/docs/components-and-props.md [hunk 6]
```diff
@@ -233,30 +233,30 @@ function Comment(props) {
...
 ```
 
-React is pretty flexible but it has a single strict rule:
+React juda moslashuvchan lekin bitta qat'iy qoidasi bor:
 
-**All React components must act like pure functions with respect to their props.**
+**Hamma React komponentlari kiritmalariga nisbatan toza funksiya sifatida yozilishi kerak**
 
-Of course, application UIs are dynamic and change over time. In the [next section](/docs/state-and-lifecycle.html), we will introduce a new concept of "state". State allows React components to change their output over time in response to user actions, network responses, and anything else, without violating this rule.
+Albatta, ilova UI'lari dinamik hisoblanadi va vaqt o'tgan sari o'zgarishi mumkin. [Keyingi bo'limda](/docs/state-and-lifecycle.html), biz "holat" (state) mavzusiga kirish qilamiz. Holat React komponentlariga vaqt o'tgan sari foydalanuvchi harakatlari, tarmoq javoblari, va boshqa holatlar orqali bu qoidani buzmagan holda reaksiya qilib o'zining qaytarmalarini o'zgartirishga imkon beradi.
```
Comment (doniyor2109) [inline content/docs/components-and-props.md]:
```suggestion
Albatta, ilova UI'lari dinamik hisoblanadi va vaqt o'tgan sari o'zgarishi mumkin. [Keyingi bo'limda](/docs/state-and-lifecycle.html), biz "holat" (state) tushunchasi bilan tanishib chiqamiz. Holat React komponentlariga vaqt o'tgan sari foydalanuvchi harakatlari, tarmoq javoblari, va boshqa holatlar orqali bu qoidani buzmagan holda reaksiya qilib o'zining qaytarmalarini o'zgartirishga imkon beradi.
```

**Review by doniyor2109 — CHANGES_REQUESTED** [review]

**Comment by zrakhimov** [comment]
Thanks @doniyor2109  for the suggestions! I'm good with them.

**Review by avocadowastaken — APPROVED** [review]

**Comment by avocadowastaken** [comment]
@zrakhimov thank you for yet another contribution :)

**Pull request merged**